stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -1457,6 +1457,7 @@ Condition number of the matrix `M`, computed using the operator `p`-norm. Valid
 function cond(A::AbstractMatrix, p::Real=2)
     if p == 2
         v = svdvals(A)
```

Comment:
```
        ifempty(A) && return float(zero(eltype(A)))
```

Reply:
At least `zero(float(real(eltype(A)))`, but I wasn't exactly sure whether that  is entirely correct either, that's why I opted to call `svdvals` unconditionally. I don't think making `cond(A::AbstractMatrix{BigFloat})` work only for empty `A` is too useful. Nevertheless, avoiding the call to `svdvals` seems like a good idea, so if someone can confirm the type of the zero to return, I'd adjust this. Is it `float(real(eltype(A)))`? Or maybe `real(eigtype(elytpe(A)))`?

Reply:
I looked up in `svdval` the the version with `eigtype`, which would provide type stability.
BTW, `cond` for `BigFloat` matrices works fine for all dimensions.

Reply:
`real(eigtype(elytpe(A)))` looks the best and does what is needed!